rust-bins/src/bin/keygen.rs

```diff
@@ -122,17 +122,28 @@ fn output_possibly_encrypted<X: SerdeSerialize>(
     fname: &PathBuf,
     data: &X,
 ) -> Result<(), std::io::Error> {
-    let pass = rpassword::read_password_from_tty(Some(
-        "Enter password to encrypt credentials (leave empty for no encryption): ",
-    ))?;
-    if pass.is_empty() {
-        println!("No password supplied, so output will not be encrypted.");
-        write_json_to_file(fname, data)
-    } else {
-        let plaintext = serde_json::to_vec(data).expect("JSON serialization does not fail.");
-        let encrypted =
-            crypto_common::encryption::encrypt(&pass.into(), &plaintext, &mut rand::thread_rng());
-        write_json_to_file(fname, &encrypted)
+    loop {
+        let pass = rpassword::read_password_from_tty(Some(
+            "Enter password to encrypt credentials (leave empty for no encryption): ",
+        ))?;
+        if pass.is_empty() {
+            println!("No password supplied, so output will not be encrypted.");
+            return write_json_to_file(fname, data);
+        } else {
+            let pass2 = rpassword::read_password_from_tty(Some("Re-enter password: "))?;
+            if pass != pass2 {
+                println!("Passwords were not equal. Try again.");
+            } else {
+                let plaintext =
+                    serde_json::to_vec(data).expect("JSON serialization does not fail.");
+                let encrypted = crypto_common::encryption::encrypt(
+                    &pass.into(),
+                    &plaintext,
+                    &mut rand::thread_rng(),
+                );
+                return write_json_to_file(fname, &encrypted);
+            }
+        }
```

Comment:
I think it would be good to factor this process into a function

```rust
/// Ask for a password and a confirmation ...
fn ask_for_password_confirm(prompt: &str) -> String {
    ...
}
```
possibly with a few more parameters.